web/app.js

```diff
@@ -2459,6 +2468,13 @@ if (typeof PDFJSDev === "undefined" || PDFJSDev.test("GENERIC")) {
 function webViewerPresentationMode() {
   PDFViewerApplication.requestPresentationMode();
 }
+function webViewerSwitchAnnotationEditorMode(evt) {
+  if (evt.toggle) {
+    PDFViewerApplication.pdfViewer.annotionEditorEnabled = true;
+  } else {
+    PDFViewerApplication.pdfViewer.annotationEditorMode = evt.mode;
+  }
```

Comment:
Sorry, but why?
This really ought to be simply `PDFViewerApplication.pdfViewer.annotationEditorMode = evt.mode;` and nothing else here.

Reply:
Oh, ok I think I understand what you want.
So to sum up and to be sure that we're ok:
 - clicking on `select`:
   * when off, switch to on and set tool buttons to off
   * when on, switch to off and set tool buttons to off
 - when clicking on a tool button:
   * when off, switch to one and set other tool buttons to off
   * when on do nothing

Finally, each tool button (FreeText, Ink, ....) is equivalent to a radio button.
Is it correct ?

Reply:
The idea was actually that *all* buttons behave as if they were radio-buttons, including the `editorNone`-one (hence no special handling for that one).

Basically, it's only possible to have *one* of the None/FreeText/Ink/... buttons toggled at any one time. Does that help?